src/component/handlers/edit/editOnPaste.js

```diff
@@ -87,9 +96,37 @@ function editOnPaste(editor: DraftEditor, e: SyntheticClipboardEvent): void {
     }
   }
 
-  let textBlocks: Array<string> = [];
   const text = data.getText();
-  const html = data.getHTML();
+  let html = getHTML(data);
+
+  if (text && !html) {
+    // The pasted content has text, but not HTML. For certain browsers (old versions of Safari, IE, and Edge)
+    // the html isn't provided as part of the clipboardData. To work around this, follow the following algorithm:
+    // Do NOT call e.preventDefault(). Instead, we want the browser to paste, just not in the editor element.
+    // Instead, move focus to a dummy contentEditable div (the pasteTrap),
+    // let the paste event through, and then copy the html out of the paste trap.
+    // It is important to call setMode('paste') to disable draft's event handlers (so it is blisfully unaware)
```

Comment:
Minor: Same comment as earlier, we shouldn't talk about Draft as if it were a 3rd party